### PR TITLE
feat: useSelector-prefer-selectors

### PR DESCRIPTION
**Pull request by martini97**
Since redux added hooks I thought it would be nice to have a rule to enforce selectors on the `useSelector` hook.

### CODE REVIEW

File: tests/lib/rules/useSelector-prefer-selectors.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      options: [{
+        matching: '^selector$',
+      }],
+    },
+  ],
+  invalid: [{
+    code: 'const property = useSelector((state) => state.x)',
+    errors: [
+      {
+        message: 'useSelector should use a selector function.',
```
Comment (DianaSuvorova) [inline tests/lib/rules/useSelector-prefer-selectors.js:42]: @martini97  Thanks a lot for the PR. 

Would it be possible to rephrase this error. I believe `(state) => state.x` is a function technically.
I guess it is not a named function it is anonymous function.  Not sure if there is a better term to describe this error.

File: docs/rules/useSelector-prefer-selectors.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#  Enforces that all useSelector properties use selector functions. (react-redux/useSelector-prefer-selectors)
```
Comment (DianaSuvorova) [inline docs/rules/useSelector-prefer-selectors.md:1]: `Enforces that all useSelector properties use selector functions.` -- > `Enforces that all useSelector hooks use named selector functions` That  is how I understood the rule you are proposing. Please correct me if I am wrong

File: docs/rules/useSelector-prefer-selectors.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+Using selectors in `useSelector` to pull data from the store or [compute derived data](https://redux.js.org/recipes/computing-derived-data#composing-selectors) allows you to uncouple your containers from the state architecture and more easily enable memoization. This rule will ensure that every prop utilizes a selector.
```
Comment (DianaSuvorova) [inline docs/rules/useSelector-prefer-selectors.md:3]: uncouple --> decouple

(sorry for being picky :) )

Comment (DianaSuvorova) [reply docs/rules/useSelector-prefer-selectors.md:3]: again, I think that anonymous functions are technically selectors as well. The rule enforces that these selectors are defined separately as apposed inline. Would be nice to clarify that, if that is correct.

**Comment by martini97** [comment]
Thanks for the review @DianaSuvorova , I have applied your suggestions.

> (sorry for being picky :) )

No problem :)

**Comment by DianaSuvorova** [comment]
@martini97  Beautiful! Thank you for your contribution to the better use of redux with hooks!

**Pull request merged**